File: src/lib/api/notes.ts
import { type Note, commands } from "@/lib/bindings";

export type { Note };

export async function listNotes(): Promise<Note[]> {
  const result = await commands.listNotes();
  if (result.status === "error") throw new Error(JSON.stringify(result.error));
  return result.data;
}

export async function createNote(title: string, body: string): Promise<Note> {
  const result = await commands.createNote(title, body);
  if (result.status === "error") throw new Error(JSON.stringify(result.error));
  return result.data;
}

export async function deleteNote(id: number): Promise<void> {
  const result = await commands.deleteNote(id);
  if (result.status === "error") throw new Error(JSON.stringify(result.error));
}

File: src-tauri/src/commands/notes.rs
use sqlx::SqlitePool;
use tauri::State;

use crate::error::AppResult;
use app_core::domain::notes::Note;

/// RS-09 のサンプルコマンド。SQL 組み立てとコンパイル時検証は `app_core::domain::notes` 側
/// （`sqlx::query!`）に閉じ、ここでは呼び出しと State からの取得のみを行う。
#[tauri::command]
#[specta::specta]
pub async fn create_note(
    pool: State<'_, SqlitePool>,
    title: String,
    body: String,
) -> AppResult<Note> {
    Ok(app_core::domain::notes::create(&pool, &title, &body).await?)
}

#[tauri::command]
#[specta::specta]
pub async fn list_notes(pool: State<'_, SqlitePool>) -> AppResult<Vec<Note>> {
    Ok(app_core::domain::notes::list(&pool).await?)
}

#[tauri::command]
#[specta::specta]
pub async fn delete_note(pool: State<'_, SqlitePool>, id: i64) -> AppResult<()> {
    Ok(app_core::domain::notes::delete(&pool, id).await?)
}
